src/logic/calculate.js

```diff
@@ -0,0 +1,70 @@
+import operates from './operate';
+
+const calculate = (calcData, btnName) => {
+  let {
+    total,
+    next,
+    operate,
+  } = calcData;
+
+  const operators = ['+', '-', 'x', '÷'];
+  const numbers = ['0', '1', '3', '4', '5', '6', '7', '8', '9'];
+
+  switch (btnName) {
+    case 'AC':
+      total = null;
+      next = null;
+      operate = null;
+      break;
+
+    case '+/-':
+      if (total) {
+        total *= -1;
+      }
+      if (next) {
+        next *= -1;
+      }
+      break;
+
+    case '%':
+      if (total) {
+        next = (total * 0.01).toString;
+      }
+      break;
```

Comment:
- [ ] Kindly remove this `%` case, it should be handled in `src/logic/operate.js` which you've already rightfully done.